docs/rfcs/0003/src/fence_hpe_hss_ssh/fence_hpe_hss_ssh.py

```diff
@@ -0,0 +1,138 @@
+#!/usr/bin/python3
+# -*- coding: utf-8 -*-
+#
+# Copyright 2021, 2022 Hewlett Packard Enterprise Development LP
+# Other additional copyright holders may be indicated within.
+#
+# The entirety of this work is licensed under the Apache License,
+# Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License.
+#
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import sys, re, time
+import atexit
+from os.path import exists
+
+sys.path.append("@FENCEAGENTSLIBDIR@")
+
+# In some cases, we do not wish to allow power off
+allow_powerOff = False
+
+try:
+	from fencing import *
+	from fencing import fail, EC_STATUS
+except:
+	if exists("/usr/share/fence/fencing.py"):
+		sys.path.append("/usr/share/fence/")
+		try:
+			from fencing import *
+			from fencing import fail, EC_STATUS
+		except:
+			fence_import_error()
+	else:
+		fence_import_error()
+		
+
+def fence_import_error():
+	print("Error importing fencing libs, may need to update python path")
+	print("e.g. export PYTHONPATH=/usr/share/fence/")
+	exit(1)
+
+def get_power_status(conn, options):
+	"""
+	Checks the power state of node 0
+
+	Returns:
+	"on" - Node 0 is powered on
+	"off" - Any power state other than ON
+	"""
+
+	# TODO: Perhaps drive this off of pmck_host_map
+	conn.send_eol("redfish node status | grep 'node 0'")
+
+	re_state = re.compile('\[(.*)', re.IGNORECASE)
+	conn.log_expect(re_state, int(options["--shell-timeout"]))
+
+	status = conn.match.group(1).lower()
+
+	if status.startswith("on"):
+		return "on"
+	else:
+		return "off"
+
+def set_power_status(conn, options):
+	"""
+	Sets the power state of Node 0 to 'on' or 'off'
+	"""
+
+	if options["--action"] == "on":
+		conn.send_eol("redfish node 0 on")
+	else:
+		if allow_powerOff:
+			conn.send_eol("redfish node 0 forceoff")
```

Comment:
This might be a Tony question: Does this also turn off the rabbit-s?

Reply:
Yes

Reply:
No, only the rabbit node.  There is a different HSS for the S.